### DISCUSSION TITLE
Is there a way to merge action arguments into state data?

### DISCUSSION THREAD

**Discussion by giorgosera** [serverlessworkflow/specification#759] [Q&A] [answered]
My use case is that I want to be able to keep track of each actions inputs and outputs in the workflow state. I'm using Temporal fir workflow runtime and I want to be able to query the internal workflow state and get the inputs and outputs of a specific state action.

We can easily do this for the results of an action but I was wondering if the spec can support the same for the arguments.

**Comment by ricardozanini** [comment, marked as answer]
@giorgosera I believe that's more of a runtime implementation concern. For instance, using an audit component, you can keep track of the ins and outs of the tasks/states. The DSL already has well-established borders across the tasks, so a runtime could, for instance, print logs of the current data state for each transition.

**Comment by cdavernas** [comment]
Synapse easily achieves that! As @ricardozanini said, this is I believe but a runtime concern. 

On a side note, logging and/or export can be easily achieved using [extensions](https://github.com/serverlessworkflow/specification/blob/main/dsl-reference.md#extension) 😉
